tools/scripts/stm32.mk

```diff
@@ -0,0 +1,95 @@
+CFLAGS += -std=gnu11 \
+	-g3 \
+	-DUSE_HAL_DRIVER \
+	-DDEBUG \
+	-O0 \
+	-ffunction-sections \
+	-fdata-sections \
+	--specs=nano.specs \
+	-mfpu=fpv4-sp-d16 \
+	-mfloat-abi=hard \
+	-mthumb \
+	-mcpu=cortex-m4
+
+LDFLAGS	= -mcpu=cortex-m4 \
+	--specs=nosys.specs \
+	-Wl,--gc-sections \
+	-static \
+	--specs=nano.specs \
+	-mfpu=fpv4-sp-d16 \
+	-mfloat-abi=hard \
+	-mthumb \
+	-Wl,--start-group \
+	-lc \
+	-lm \
+	-Wl,--end-group
+
+CC = arm-none-eabi-gcc
+AS = arm-none-eabi-gcc
+AR = arm-none-eabi-ar
+
+rwildcard=$(foreach d,$(wildcard $(1:=/*)),$(call rwildcard,$d,$2) $(filter $(subst *,%,$2),$d))
+ifneq '' '$(call rwildcard,src,stm32f4*)'
+SUBPLATFORM = stm32f4
+CFLAGS += -I$(NO-OS)/libraries/stm32/STM32CubeF4/Drivers/STM32F4xx_HAL_Driver/Inc \
+	-I$(NO-OS)/libraries/stm32/STM32CubeF4/Drivers/STM32F4xx_HAL_Driver/Inc/Legacy \
+	-I$(NO-OS)/libraries/stm32/STM32CubeF4/Drivers/CMSIS/Device/ST/STM32F4xx/Include \
+	-I$(NO-OS)/libraries/stm32/STM32CubeF4/Drivers/CMSIS/Include
+TARGETCFG = target/stm32f4x.cfg
+else
+$(error Couldn't detect stm32 family.$(ENDL))
+endif
+
+openocd_paths:
+ifndef OPENOCD_SCRIPTS
+	$(error OPENOCD_SCRIPTS not found in shell environment.$(ENDL))
+endif
+
```

Comment:
If you have this extra line, $(error OPENOCD_BIN not found in shell environment.$(ENDL))
will show when running the makefile. not for the targer openocd_paths

Reply:
Not sure if I understand the comment. It only tests it when PLATFORM is stm32 and rule is debug.

Reply:
oh ok, the empty line should not be there, got it ! will fix

Reply:
will remove the empty line !